### ISSUE TITLE
[Proposal] Add useHelmSecrets option to helm config

### ISSUE THREAD

**Issue by RoryShively** [GoogleContainerTools/skaffold#1178]
**I'm rewriting this because it hasn't received enough attention. I originally had proposed to add plugins to the helm config but my proposed implementation was quite complicated. This new proposal has been greatly simplified**

I created a PR for helm-secrets that will make integration with skaffold easier.
https://github.com/futuresimple/helm-secrets/pull/67

We can create a config option that will use helm-secrets if set to true to decrypt secrets on the fly before deployment. The config might look like this:
```
apiVersion: skaffold/v1alpha4
kind: Config
build:
  ...
deploy:
  helm:
    releases:
    - name: helm-secrets-test
      chartPath: helm_secrets_test
      valuesFiles:
      - helm_secrets_test/helm_vars/values.yaml
      - helm_secrets_test/helm_vars/secrets.yaml
      useHelmSecrets: true
```

All that needs to change in the skaffold code is `pkg/skaffold/deploy/helm` line 110
```
	args := append([]string{"secrets", "--kube-context", h.kubeContext}, arg...)
```
Add secrets as an argument if useHelmSecrets is true

**Comment by RoryShively** [comment]
I created a PR for helm-secrets that will make integration with skaffold easier.
https://github.com/futuresimple/helm-secrets/pull/67

All that needs to change in the skaffold code is pkg/skaffold/deploy/helm line 110
```
	args := append([]string{"secrets", "--kube-context", h.kubeContext}, arg...)
```
Adding secrets to the args


As long as the user has helm-secrets from this (fork)[https://github.com/RoryShively/helm-secrets/tree/skaffold] (pending merge) then everything will run as expected with on-the-fly decryption. I can make a PR to put this in the config if there's interest

Maybe something like this?
```
apiVersion: skaffold/v1alpha4
kind: Config
build:
  ...
deploy:
  helm:
    releases:
    - name: helm-secrets-test
      chartPath: helm_secrets_test
      valuesFiles:
      - helm_secrets_test/helm_vars/values.yaml
      - helm_secrets_test/helm_vars/secrets.yaml
      useSecrets: true
```

**Comment by nkubala** [comment]
I'm a +1 on this. @RoryShively would you like to send a PR?

**Comment by RoryShively** [comment]
@nkubala Yeah. Sorry about the late reply, I'll get you something by the end of the day hopefully

**Comment by RoryShively** [comment]
@nkubala Just submitted a first draft
https://github.com/GoogleContainerTools/skaffold/pull/1617

- Placed useHelmSecrets option in latest schema
- prepended "secrets" to args for helm deployment if useHelmSecrets selected

**Comment by Ps3ud-0nym** [comment]
When using this, it works well but doesn't seem to delete the decrypted secrets file automatically after deploy. Could someone look at this please?

**Comment by nkubala** [comment]
this should be fixed through #1617, thanks @RoryShively!
